backend/src/crd/controller/scheduledworkflow/util/cron_schedule.go

```diff
@@ -15,13 +15,13 @@
 package util
 
 import (
-	"math"
 	"time"
 
 	swfapi "github.com/kubeflow/pipelines/backend/src/crd/pkg/apis/scheduledworkflow/v1beta1"
 	wraperror "github.com/pkg/errors"
 	"github.com/robfig/cron"
 	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // CronSchedule is a type to help manipulate CronSchedule objects.
```

Comment:
Hi @NikeNano, after taking a look at the big picture, I start to think location should be a field of `CronSchedule`, because only CronSchedule cares about timezone. If you add it as field of `CronSchedule`, you can save the vast spreading change of adding an argument to `GetNextScheduledEpoch` methods.

Reply:
I believe the periodic scheduling also suffers from that it adheres to the UTC timezone only? If we keep it on the scheduler it will be easier to also upgrade the periodic scheduler as well. WDYT?

Reply:
I believe I have asked about this before, but I still do not understand. The periodic schedule only schedules for start time, start time + delta, start time + delta * 2, ... Why is it affected by timezone?

Reply:
I might be missing something but since there is no awareness of timezone for periodic jobs, the start and endtime options will be based upon UTC as I understand, thus these will be adjusted to UTC which have to be accounted for by the user.

Reply:
@nikenano I think user's browser will be responsible for converting timezone specific settings to start time and end time in UTC before passing them to backend. Is there anything wrong in that process? If so, we should fix the UI code instead I think.

Reply:
Ohh missed that! No I don't think so. I will move the location as you suggested.